### ISSUE TITLE
SOCKS5 Proxy/Orbot Support

### ISSUE THREAD

**Issue by Woman-at-arms** [ImranR98/Obtainium#121]
Many package managers and other apps have proxy configuration support.

This is especially useful for guarding package managers against MITM when paired with tor.

See how Kicksecure/Whonix and Qubes updates work for example.

I would love to see SOCKS5 proxy or native Orbot support in Obtainium!

**Comment by ImranR98** [comment]
An HTTP proxy option can definitely be added. Not familiar with SOCKS5 though. For Orbot, doesn't it act as a VPN that any app's traffic can be passed through?

**Comment by Woman-at-arms** [comment]
Orbot supports both proxy and VPN modes. The default is proxy mode, where it runs a local SOCKS5 proxy.

SOCKS5 is the most modern type of HTTP proxy and the one that Orbot supports.

Proxy mode for Orbot is generally preferable, since you can use it when you already have a VPN.

**Comment by kyclol** [comment]
+1 

this option is needed to be able to download from tor/i2p git repos

**Comment by ThePhoDit** [comment]
This feature would be great IMO. It would work with already present services like custom F-droid repos or web scraping

**Comment by Solomon1732** [comment]
> An HTTP proxy option can definitely be added. Not familiar with SOCKS5 though. For Orbot, doesn't it act as a VPN that any app's traffic can be passed through?

@ImranR98 I don't know if you read what I said above, about Orbot also supporting HTTP proxy?

**Comment by J0J0XMR** [comment]
SOCKS5 is definitely the way to go, ideally set per-repo instead of globally for entire app. 

That way regular repos wouldn't need to be downloaded through proxy, which will make the download much slower.

@ImranR98 Is this planned anytime soon? So many people would love this!

**Comment by E021ntox** [comment]
Still no http proxy support?

**Comment by CKMz17** [comment]
> SOCKS5 is definitely the way to go, ideally set per-repo instead of globally for entire app. 
> 
> That way regular repos wouldn't need to be downloaded through proxy, which will make the download much slower.
> 
> @ImranR98 Is this planned anytime soon? So many people would love this!

Yes, please. If possible, implement this both globally and per repository
